### ISSUE TITLE
how to get ssl working to support https

### ISSUE THREAD

**Issue by zs787** [heroku/heroku-buildpack-nginx#18]
Hi,

I am using this build pack with my application in Heroku and was just wondering if there is a way to make it work with ssl ? 

As it seems it was not originally compiled with the ssl module
log message I got: ... the "ssl" parameter requires ngx_http_ssl_module in ./config/nginx.conf:30

~ $ nginx -V
nginx version: nginx/1.9.5
built by gcc 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.10) 
configure arguments: --with-pcre=pcre-8.37 --with-zlib=zlib-1.2.11 --with-http_gzip_static_module --with-http_realip_module --prefix=/tmp/nginx --add-module=//tmp/nginx.lcuziO0aKW/nginx-1.9.5/headers-more-nginx-module-0.261

I tried to modify the build script and add the module but doesn't seem to work.
--with-http_ssl_module \

Thank you

**Comment by beanieboi** [comment]
hey,

adding `--with-http_ssl_module \` on line 50 (e.g.) should work.
Can you explain what are you trying to achieve?

ben

**Comment by beanieboi** [comment]
are you in common runtime or private spaces? in common runtime there is no way to accept SSL connection at your dyno, it will be handled by the Heroku Loadbalancer that terminates the SSL for you. there is also my private buildpack, which is a enhanced version of the heroku buildpack. it includes a more recent version of nginx and more modules.

https://github.com/beanieboi/nginx-buildpack

**Comment by zs787** [comment]
Thank you and I appreciate your reference, looks like I will use it instead.
As for my application its actually in a shield space/private space and planning to use internal routing.
Currently its failing as I need to `proxy_pass https://...` but wouldn't do it.

Just out of curiosity I would like to know if what I did last time was the proper way, I forked this Heroku build-pack and modified the script by adding `--with-http_ssl_module` which did not work. I think it needs to be done for the build-pack repo and not in my nginx app repo (the forked one), as the build pack will overwrite it.
